### ISSUE TITLE
Contentful image JSON object now allowing access to keys

### ISSUE THREAD

**Issue by KyleTryon** [KyleTryon/VidFireBlog#1]
https://github.com/KyleTryon/VidFireBlog/blob/master/pages/blog/_slug.vue

When attempting to retrieve the image object from Contentful, I am somehow unable to access anything inside the object.

I am currently displaying the object on the page as a means to console log. The following line will display the image object.

```
<span>{{article.fields.headerImage}}</span>

```

output:

```
{ sys:
   { space: { sys: [Object] },
     id: '4stFqnjaLeu66OKU2gYykG',
     type: 'Asset',
     createdAt: '2017-12-17T19:27:41.461Z',
     updatedAt: '2017-12-17T19:27:41.461Z',
     revision: 1,
     locale: 'en-US' },
  fields:
   { title: 'MicrophoneTitle',
     file:
      { url: '//images.contentful.com/6lqnknawo3ew/4stFqnjaLeu66OKU2gYykG/472ef53880c5296ee6096864b2e5fc08/MicrophoneTitle.jpg',
        details: [Object],
        fileName: 'MicrophoneTitle.jpg',
        contentType: 'image/jpeg' } } }

```

That will print out the object I listed above. But If I change that code to anything inside the object I get a type error

```

<span>{{article.fields.headerImage.fields}}</span>

```

`TypeError
Cannot read property 'fields' of undefined`

error: https://i.imgur.com/F35pMlQ.png?1

**Comment by KyleTryon** [comment]
Just proof of correct json string

https://i.imgur.com/81Jjzya.png

**Comment by KyleTryon** [comment]
More tests I have done.

```
export default {
    props: ['article'],
    computed: {
      image: function () {
        return this.article.fields.headerImage
      }
    },
```
So this is how I am cheating (trying to)
I have this on the page `<span>{{image}}</span>`
Which is currently spitting out this. And it is on the page.

`{ "sys": { "space": { "sys": { "type": "Link", "linkType": "Space", "id": "6lqnknawo3ew" } }, "id": "4stFqnjaLeu66OKU2gYykG", "type": "Asset", "createdAt": "2017-12-17T19:27:41.461Z", "updatedAt": "2017-12-17T19:27:41.461Z", "revision": 1, "locale": "en-US" }, "fields": { "title": "MicrophoneTitle", "file": { "url": "//images.contentful.com/6lqnknawo3ew/4stFqnjaLeu66OKU2gYykG/472ef53880c5296ee6096864b2e5fc08/MicrophoneTitle.jpg", "details": { "size": 41478, "image": { "width": 900, "height": 330 } }, "fileName": "MicrophoneTitle.jpg", "contentType": "image/jpeg" } } }`
Cool! To be safe. I edited a comment and saved it again to mnake it regenerate the page. looks like its still there. i'll do it one more time.. We're good.

Modifying image to this, the first key in that object.
```

computed: {
      image: function () {
        return this.article.fields.headerImage.sys
      }
```

Cannot read property 'sys' of undefined. image: https://cdn.discordapp.com/attachments/348033813873688578/395443574709157892/unknown.png

Ok. So maybe. Somehow. This is a string? So. I test for that.

```
computed: {
      image: function () {
        let img = this.article.fields.headerImage
        try {
          return JSON.parse(img)
        }
        catch(e) {
          console.log(e)
          return img
        }
      }
    },
```
logs out the object again... but I can already see an error in console

```
SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at VueComponent.image (pages/blog/index.d81e2487d058a9aff27b.js:371:21)
    at Object.get (//VidFireBlog/node_modules/vue/dist/vue.runtime.common.js:1654:20)
     ... goes on
```

Not sure what's going on there... attempted to add .sys anyway and of course it is undefined.

<span>{{image.sys}}</span>

There is something malformed about this JSON object somehow.

**Comment by LeheneTudor** [comment]
Did you find any workaround for this? I also have this key accessing issue for the image URL. Really strange behavior... :|

**Comment by LeheneTudor** [comment]
Ok I found out why this was happening.

In my case I had multiple "items" with file URLs (images) and one of them did not have a published image which, obviously, was undefined. 

I set the image field in contentful as "required" but, somehow, I am able to save the item even if it doesn't have a required field filled. I am new to contentful so maybe I'm missing something but I hope this will help other dummies like me solve the items.fields.file undefined error in contentful.
